tests/integration/checkpoint.bats

```diff
@@ -252,7 +252,7 @@ function simple_cr() {
 	# in time when the last page is lazily transferred to the destination.
 	# Killing the CRIU on the checkpoint side will let the container
 	# continue to run if the migration failed at some point.
-	[ -n "$RUNC_USE_SYSTEMD" ] && set_cgroups_path
+	[ -v RUNC_USE_SYSTEMD ] && set_cgroups_path
```

Comment:
I'm not familiar with the `-v` option (tried searching the man page, but couldn't find it 😅); what does it do? Is it something Bash specific? (in which case perhaps this one should explicitly use `[[`) 

If it's to differentiate between "var not set" and "var set, but empty", I'm wondering if it's more error-prone 🤔 (maybe not, not sure, would have to think of various scenarios)

Reply:
help [
help [[

It tells if the variable is set.

It is bash specific, but we use bats, which is already bash-specific, and have /bin/bash shebang everywhere.

[ and [[ have slightly different rules and both support -v in bash.

The alternative, also bash-specific, is something like `[ -n "${VAR:-x}" ]`, which looks way more cryptic than -v.

Reply:
Interesting; neither `help [` / `help [[` nor `man [` / `man test` (`man [[` returns an error on Ubuntu) showed it (at least, couldn't find it).

Reply:
So my concern is a bit that `-v ENABLE_FOO` only checks if it's set, but can be confusing if (e.g.) `ENABLE_FOO=0`

```bash
(set -eu; [ -v ENABLE_FOO ] && echo "FOO is enabled"; echo $?)
1

(set -eu; ENABLE_FOO=0; [ -v ENABLE_FOO ] && echo "FOO is enabled"; echo $?)
FOO is enabled
0

(set -eu; ENABLE_FOO=1; [ -v ENABLE_FOO ] && echo "FOO is enabled"; echo $?)
FOO is enabled
0
```

From that perspective, `[ "${ENABLE_FOO}" = "1" ]` (e.g.) feels more transparent (and possibly less error prone?)
But if we really need `-u` everywhere, that would probably need some initialization code to be sure variables
are set to a default, like

```bash
: "${ENABLE_FOO:=0}"
```

(which we do for `$ROOTLESS_FEATURES` I see)

Reply:
Yes, what you propose is an alternative to using `-v`, and is sometimes used.

I guess what we need to do is document all those variables. For most of them, any value means its "on/true/1/yes/enabled", and the unset variable means "off/false/0/no/disabled".

The only exception is `CGROUP_UNIFIED`, which is either `yes` or `no` (or unset). When I think about it, it makes sense to change this to `CGROUP_V1` and `CGROUP_V2`, so we treat these the same as all the others (unset means no, set means yes).

Reply:
Since I am already doing some changes with `CGROUP_UNIFIED` in #3369, let's replace it with `CGROUP_V2` and `CGROUP_V1` in there, and keep this PR focused on one single thing -- enabling `set -u`.

Reply:
https://github.com/opencontainers/runc/pull/3369/commits/14445d5edf05fef2aeeb6bdeb649afdaf6629bf9

Reply:
IOW this is addressed in #3369

Reply:
I agree that `-v VAR` is very subtly different from `-n $VAR` -- when I make these conversions, I usually just use `:-` (or `-`) such that it doesn't trip `set -u` (unless it's used multiple times, then I use the `: "${VAR:=}"` as suggested by Sebastiaan):

```suggestion
	[ -n "${RUNC_USE_SYSTEMD:-}" ] && set_cgroups_path
```

It looks more cryptic, but it matches the old behavior (where both empty _and_ unset are considered "false" -- in many cases/places, it's trivial to set a variable to false and impossible to unset it, like Docker's `--env` for example).

Reply:
@tianon thanks for your input! 

I am very open to use '${VAR:-}` where appropriate, but to me it seems that `-v VAR` is a tad cleaner, because it is more readable, and readability is a big factor in my book.

Note that I only switch to `-v` in cases where the flag is essentially a binary 0/1, since set/unset fits it better than
- empty/non-empty (because we do not have to provide a kludge to satisfy `set -u`),
- "yes"/"no", 1/0 etc. (because string or numeric values are not standardized anywhere in the code, and we don't have to set the "no"/0 case).

I can certainly rework the whole set again to use `-n|-z "${VAR:-}"` instead of `[!] -v VAR`, if I clearly see its benefits.

You point out to two things
1. Matching the old behavior. I agree, but it is not an issue here since this is an isolated codebase which we fully control, so we can convert it all in one sitting (which is what this PR does) and won't add any new bugs (well, hopefully).
2. Inability to unset a variable when using Docker. Also true, but I don't see this is practically an issue (for the same reason -- this is an isolated set of tests which we run in some specific environments we fully control).

My third reason is sheer laziness -- it took a while to convert to `-v`, and I have another PR which is based on this one (brining in some more cleanups). I can't convince myself to spend a few hours to rework this (and that PR) one more time.

All it all, I'd like to leave this as is, unless there's a strong objection. I am listening and can definitely reconsider.

Reply:
Totally fair 👍